src/Servers/Kestrel/Core/src/Internal/Http/Http1OutputProducer.cs

```diff
@@ -197,13 +213,20 @@ public void Advance(int bytes)
                     return;
                 }
 
-                if (_autoChunk)
+                if (!_startCalled)
                 {
-                    if (bytes < 0)
+                    if (bytes >= 0)
                     {
-                        throw new ArgumentOutOfRangeException(nameof(bytes));
-                    }
+                        if (_currentSegment.Length < _position + bytes)
```

Comment:
This can overflow

Reply:
_sigh_

Reply:
What's our recommendation for checking overflows? Do we ever use checked? https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/checked

Reply:
`_currentSegment.Length - bytes < _position`

Reply:
We should probably do something similar for _advancedBytesForChunk